docs/ci-cd-integration/deploy-from-github.md

```diff
@@ -102,7 +102,98 @@ When you commit files to your project's repo on the "main" branch, Github Action
+Alternatively, you can even build the Docker image on Github and just deploy the built artifact to your CapRover instance. This will help as it does not consume RAM and CPU from your CapRover instance to build your image.
+
+In order to achieve this we will need to take the following steps to build the Docker image using GitHub Actions, store it using GitHub Packages, and then deploy it to CapRover.
+
+#### Create a GitHub Personal Access Token
+You will need to create a GitHub Personal Access Token with write permission for packages.
+
+GitHub has a great guide on creating a personal access token if you have not before. Here is the link: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token
+
+#### Create a New App
+If you do not have an app already on CapRover, create one using the instructions [here](### Create a new App)
+
+If you do already have an app on CapRover you can skip this step. 
+
+#### Enable App Token
+If you do not already have an app token for your app, create one using the instructions [here](### Enable App Token)
+
+If you do have an app token, keep it handy as we will need it in the next step. 
+
+#### Add The GitHub Secrets
+You will need to add the following information into GitHub Secrets:
+- App Name: Name of the app in CapRover
+- App Token: App token we got in the previous step
+- CapRover Server URL: URL of your CapRover Server
+- GitHub Token: GitHub Personal Access Token you created in previous step
+
+You can add GitHub Secrets using the instructions [here](### Add the Github Secrets)
+
+#### Add the Captain Definition File
+Add the `captain-definition` file described [here](### Add files to project)
+
+#### Create the GitHub Action
+GitHub Actions is the CI/CD pipeline built into GitHub. If you are unfamiliar with it, it would be beneficial to learn the basics by reviewing GitHub's Understanding GitHub Actions Docs: https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions
+
+Here is an example GitHub Action that builds a docker container on each push to a pull request and deploys it to the CapRover server (good example for a development environment set up)
+
+```
+name: Build and Deploy Docker Image
+
+on: [pull_request]
+
+jobs:
+  build_and_deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v2
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+      
+    - name: Login to Container Registry
+      uses: docker/login-action@v2
+      with:
+            registry: ghcr.io
+            username: ${{ github.repository_owner }}
+            password: ${{ secrets.GH_TOKEN }}
+
+    - name: Preset Image Name
+      run: echo "IMAGE_URL=$(echo ghcr.io/${{ github.repository_owner }}/${{ github.event.repository.name }}:$(echo ${{ github.sha }} | cut -c1-7) | tr '[:upper:]' '[:lower:]')" >> $GITHUB_ENV
```

Comment:
Is this line well tested? I think previously we had this:
```
            - run: |
                  echo "IMAGE_NAME_WITH_REGISTRY=$DOCKER_REGISTRY/$IMAGE_NAME" >> $GITHUB_ENV
                  export IMAGE_NAME_WITH_REGISTRY=$DOCKER_REGISTRY/$IMAGE_NAME
                  echo "FULL_IMAGE_NAME=$IMAGE_NAME_WITH_REGISTRY:$GITHUB_SHA-gitsha" >> $GITHUB_ENV
```

If this line is well tested, I am okay with using this new line as it's simpler.

Reply:
I borrowed this line from @maietta's example: https://github.com/maietta/SvelteKit-Prisma-Foundation-Boilerplate/blob/master/.github/workflows/production.yml

It was worked well for me so far.

Open to changing it if you want to have your suggested commands instead.

Reply:
Hi @jtrugman. Glad you found my one-liner useful. The reason I had to use craft this version was to address malformed docker image names as docker images I think require lowercase tagging. By going all lowercase, it guarantied there were no issues with creating and pushing the image to Github Container Registry.